src/validation/io-ts-validator.ts

```diff
@@ -0,0 +1,35 @@
+import * as E from 'fp-ts/Either';
+import { pipe } from 'fp-ts/function';
+import * as t from 'io-ts';
+import { formatValidationErrors } from 'io-ts-reporters';
+
+type ResultType<T> =
```

Comment:
Qui, a questo punto, ti torna molto più comodo se usi una tagged union

```typescript
type ResultType<T> =
    | {
        _tag: 'error';
        errors: string;
      }
    | {
        _tag: 'success';
        res: T;
      };
```
in modo che `transform` diventi

```typescript
  transform(value: any): A | never {
    const r = validate<I, A>(this.schema, value);

    switch(r._tag) {
      case  'success':
        return r.res;
      case 'error':
        throw new BadRequestException(r.errors)
    }
  }
```

(che poi è isomorfo a `Either<string, A>`)

Reply:
mi torna il tuo ragionamento e la tua soluzione, diciamo che la poca conoscenza di fp-ts un po mi ha piantato, infatti la ho abbozzata e poi la cosa sarebbe stata rivista nel modo migliore con voi sicuramente